include/mesh/distributed_mesh.h

```diff
@@ -81,7 +81,27 @@ class DistributedMesh : public UnstructuredMesh
    * Copy and move assignment are not allowed.
    */
   DistributedMesh & operator= (const DistributedMesh &) = delete;
-  DistributedMesh & operator= (DistributedMesh &&) = delete;
+
+  /**
+   * Overloaded operator= will move contents of other_mesh to calling
+   * DistributedMesh object.
+   */
+  DistributedMesh & operator= (DistributedMesh && other_mesh);
+
+  /**
+   * Shim to call the move assignment operator for this class
+  */
+  virtual MeshBase & assign(MeshBase && other_mesh) override
+  {
+    *this = std::move(*(cast_ptr<DistributedMesh*>(&other_mesh)));
+
+    return *this;
+  }
+
+  /**
+   * Move node and elements from other_mesh to this mesh.
```

Comment:
Something here about other_mesh needing to be a DistributedMesh.

We actually could support moving nodes and elements from a ReplicatedMesh to a DistributedMesh or vice-versa, but until we do, the function signature here is misleading.